client/cmd/dexc-desktop/app_darwin.go

```diff
@@ -0,0 +1,584 @@
+//go:build darwin
+
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"runtime"
+	"runtime/debug"
+	"runtime/pprof"
+	"strconv"
+	"sync"
+	"sync/atomic"
+	"syscall"
+	"time"
+
+	"decred.org/dcrdex/client/app"
+	"decred.org/dcrdex/client/asset"
+	dexCore "decred.org/dcrdex/client/core"
+	"decred.org/dcrdex/client/mm"
+	"decred.org/dcrdex/client/rpcserver"
+	"decred.org/dcrdex/client/webserver"
+	"decred.org/dcrdex/dex"
+	"github.com/progrium/macdriver/cocoa"
+	"github.com/progrium/macdriver/core"
+	"github.com/progrium/macdriver/objc"
+	"github.com/progrium/macdriver/webkit"
+)
+
+var (
+	webviewConfig  = webkit.WKWebViewConfiguration_New()
+	width, height  = defaultWindowWidthAndHeight()
```

Comment:
Instead of the cgo approach, I think you could do something like

```Go
func defaultWindowWidthAndHeight() (width, height int) {
	frame := cocoa.NSScreen_Main().Frame()
	return int(math.Round(frame.Size.Width)), int(math.Round(frame.Size.Height))
}
```

You'd still want to limit the max values though.

Reply:
> Instead of the cgo approach, I think you could do something like

Yh, I just wanted to reuse what we already have but we can have this.